### PR TITLE
Add --searchby

### LINKED ISSUE #936: [FR] Search in package names only
#### Affected Version
yay v9.1.0 - libalpm v11.0.3

#### Issue
It would be useful to have a built-in way to search in package names only. It's a somewhat frequent request too: https://bbs.archlinux.org/viewtopic.php?id=90164, https://bbs.archlinux.org/viewtopic.php?id=54684, https://bbs.archlinux.org/viewtopic.php?id=236426, https://bbs.archlinux.org/viewtopic.php?id=92102

### PR DESCRIPTION
**Pull request by Morganamilo**
Allow using the RPC's &by= field during -Ss

Fixes #936.

### CODE REVIEW

**Comment by Jguer** [comment]
Some mention that modules were updated would be nice in the log. Possibly separating `go-alpm`'s update from the rest. Apart from that looks fine

File: depPool.go [hunk 1]
```diff
@@ -188,7 +188,7 @@ func (dp *depPool) findProvides(pkgs types.StringSet) error {
 		words := strings.Split(pkg, "-")
 
 		for i := range words {
-			results, err = rpc.SearchByNameDesc(strings.Join(words[:i+1], "-"))
+			results, err = rpc.Search(strings.Join(words[:i+1], "-"))
```
Comment (Jguer) [inline depPool.go]: In depPool wouldn't it be better to search only Name instead of NameDesc?

Comment (Morganamilo) [reply depPool.go]: I see no reason to. More results means a better chance of finding a provide. And seems as we can fit everything into a single into query there should be no speed difference.

Comment (Jguer) [reply depPool.go]: Ok, I can agree with that.

**Comment by Jguer** [comment]
Also missing completions and man page update. I'll probably create an issue on it so it's not forgotten

**Comment by boussou** [comment]
I can't believe it is impossible to limit search only to package name in AUR?
it tried using keyword "git", to many results !!

**Pull request merged**